src/main/java/appeng/hooks/TickHandler.java

```diff
@@ -286,6 +250,73 @@ public void registerCraftingSimulation(final World world, final CraftingJob craf
         }
     }
 
+    /**
+     * Simulates the current crafting requests before they user can submit them to be processed.
+     * 
+     * @param world
+     */
+    private void simulateCraftingJobs(IWorld world) {
+        synchronized (this.craftingJobs) {
+            final Collection<CraftingJob> jobSet = this.craftingJobs.get(world);
+
+            if (!jobSet.isEmpty()) {
+                final int jobSize = jobSet.size();
+                final int microSecondsPerTick = AEConfig.instance().getCraftingCalculationTimePerTick() * 1000;
+                final int simTime = Math.max(1, microSecondsPerTick / jobSize);
+
+                final Iterator<CraftingJob> i = jobSet.iterator();
+
+                while (i.hasNext()) {
+                    final CraftingJob cj = i.next();
+                    if (!cj.simulateFor(simTime)) {
+                        i.remove();
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Ready the tiles in this world
+     * 
+     * @param world
+     */
+    private void readyTiles(IWorld world) {
+        final HandlerRep repo = this.getRepo();
+        final Map<ChunkPos, Queue<AEBaseTileEntity>> worldQueue = repo.tiles.getOrDefault(world,
+                Collections.emptyMap());
+
+        for (Iterator<Entry<ChunkPos, Queue<AEBaseTileEntity>>> it = worldQueue.entrySet().iterator(); it
```

Comment:
Not a big fan of this for-loop construct, but eh... Okay

Reply:
Yeah. looks ugly but it allows us to use `iterator.remove()` and not run into CMEs or having to maintain another collection to clean it up later. I do not want to keep the empty map entry around when not necessary. And should something constantly swap chunks in and out that additional object instantiation will not really affect anything when compared to whatever else will already happen.

Just in case that question comes up at some point. At least document the thought process here.